web/core/src/jsMain/kotlin/org/jetbrains/compose/web/events/SyntheticMouseEvent.kt

```diff
@@ -0,0 +1,89 @@
+package androidx.compose.web.events
+
+import org.w3c.dom.DataTransfer
+import org.w3c.dom.DragEvent
+import org.w3c.dom.HTMLElement
+import org.w3c.dom.events.EventTarget
+import org.w3c.dom.events.MouseEvent
+import org.w3c.dom.events.WheelEvent
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent
+ */
+open class SyntheticMouseEvent(
+    target: HTMLElement,
+    nativeEvent: MouseEvent
+) : SyntheticEvent<HTMLElement>(target, nativeEvent) {
+
+    constructor(nativeEvent: MouseEvent): this(
+        target = nativeEvent.target as HTMLElement,
+        nativeEvent = nativeEvent
+    )
+
+    private val mouseEvent = nativeEvent
+
+    val altKey: Boolean = nativeEvent.altKey
+    val button: Short = nativeEvent.button
+    val buttons: Short = nativeEvent.buttons
+    val clientX: Int = nativeEvent.clientX
+    val clientY: Int = nativeEvent.clientY
+    val ctrlKey: Boolean = nativeEvent.ctrlKey
+    val metaKey: Boolean = nativeEvent.metaKey
+    val movementX: Int = nativeEvent.asDynamic().movementX as Int
+    val movementY: Int = nativeEvent.asDynamic().movementY as Int
+    val offsetX: Double = nativeEvent.offsetX
+    val offsetY: Double = nativeEvent.offsetY
+    val pageX: Double = nativeEvent.pageX
+    val pageY: Double = nativeEvent.pageY
+    val region: String? = nativeEvent.region
+    val relatedTarget: EventTarget? = nativeEvent.relatedTarget
+    val screenX: Int = nativeEvent.screenX
+    val screenY: Int = nativeEvent.screenY
+    val shiftKey: Boolean = nativeEvent.shiftKey
+    val x: Double = nativeEvent.x
+    val y: Double = nativeEvent.y
+
+    fun getModifierState(keyArg: String): Boolean = mouseEvent.getModifierState(keyArg)
+}
+
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent
+ */
+class SyntheticWheelEvent(
+    target: HTMLElement,
+    nativeEvent: WheelEvent
+) : SyntheticMouseEvent(
+    target,
+    nativeEvent
+) {
+
+    constructor(nativeEvent: WheelEvent): this(
+        target = nativeEvent.target as HTMLElement,
+        nativeEvent = nativeEvent
+    )
+
+    val deltaX: Double = nativeEvent.deltaX
+    val deltaY: Double = nativeEvent.deltaY
+    val deltaZ: Double = nativeEvent.deltaZ
+    val deltaMode: Int = nativeEvent.deltaMode
```

Comment:
Is it sort of enum?

Reply:
yes, sort of:

here is its defenition in kotlin/js
```kotlin
 companion object {
        val DOM_DELTA_PIXEL: Int
        val DOM_DELTA_LINE: Int
        val DOM_DELTA_PAGE: Int
        val NONE: Short
        val CAPTURING_PHASE: Short
        val AT_TARGET: Short
        val BUBBLING_PHASE: Short
    }
```

Reply:
it's better to use another type than `Int`, to prompt available values and use safe `when(deltaMode)`